drivers/bluetooth/hci/hci_stm32wb0.c

```diff
@@ -0,0 +1,499 @@
+/* hci_stm32wb0.c - HCI driver for stm32wb0x */
+
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/hci_types.h>
+#include <zephyr/drivers/bluetooth.h>
+#include "bleplat_cntr.h"
+#include "ble_stack.h"
+#include "stm32wb0x_hal_radio_timer.h"
+#include "miscutil.h"
+#include "pka_manager.h"
+#include "app_conf.h"
+#include "dtm_cmd_db.h"
+#include "dm_alloc.h"
+#include "aci_adv_nwk.h"
+#include "app_common.h"
+#include "hw_rng.h"
+#include "hw_aes.h"
+#include "hw_pka.h"
+
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_driver);
+
+#define DT_DRV_COMPAT st_hci_stm32wb0
+
+/* Max HS startup time expressed in system time (1953 us / 2.4414 us) */
+#define MAX_HS_STARTUP_TIME	320
+#define BLE_WKUP_PRIO		0
+#define BLE_WKUP_FLAGS		0
+#define BLE_TX_RX_PRIO		0
+#define BLE_TX_RX_FLAGS		0
+#define CPU_WKUP_PRIO		1
+#define CPU_WKUP_FLAGS		0
+#define BLE_ERROR_PRIO		3
+#define BLE_ERROR_FLAGS		0
+#define BLE_RXTX_SEQ_PRIO	3
+#define BLE_RXTX_SEQ_FLAGS	0
+#define PKA_PRIO		2
+#define PKA_FLAGS		0
+
+#define MAX_EVENT_SIZE			259
+#define MAX_ISO_DATA_LOAD_LENGTH	512
+
+#define PACKET_TYPE		0
+#define EVT_HEADER_TYPE		0
+#define EVT_HEADER_EVENT	1
+#define EVT_HEADER_SIZE		2
+#define EVT_LE_META_SUBEVENT	3
+#define EVT_VENDOR_CODE_LSB	3
+#define EVT_VENDOR_CODE_MSB	4
+
+static uint32_t __noinit dyn_alloc_a[BLE_DYN_ALLOC_SIZE >> 2];
+static uint8_t buffer_out_mem[MAX_EVENT_SIZE];
+static struct k_work_delayable hal_radio_timer_work, ble_stack_work;
+
+static struct net_buf *get_rx(uint8_t *msg);
+static PKA_HandleTypeDef hpka;
+
+#if CONFIG_BT_EXT_ADV
+static uint32_t __noinit aci_adv_nwk_buffer[CFG_BLE_ADV_NWK_BUFFER_SIZE >> 2];
+#endif /* CONFIG_BT_EXT_ADV */
+
+struct hci_data {
+	bt_hci_recv_t recv;
+};
+
+/* Dummy implementation */
+int BLEPLAT_NvmGet(void)
+{
+	return 0;
+}
+
+static void blestack_process(struct k_work *work)
+{
+	BLE_STACK_Tick();
+	if (BLE_STACK_SleepCheck() == 0) {
+		k_work_reschedule(&ble_stack_work, K_NO_WAIT);
+	}
+}
+
+static void vtimer_process(struct k_work *work)
+{
+	HAL_RADIO_TIMER_Tick();
+}
+
+/* "If, since the last power-on or reset, the Host has ever issued a legacy
+ * advertising command and then issues an extended advertising command, or
+ * has ever issued an extended advertising command and then issues a legacy
+ * advertising command, the Controller shall return the error code Command
+ * Disallowed (0x0C)."
+ * This function returns 1 if an error has to be given.
+ */
+static uint8_t check_legacy_extended_call(uint16_t opcode, uint8_t *buffer_out)
+{
+	static bool legacy_cmd_issued = FALSE, extended_cmd_issued = FALSE;
+	bool allowed = TRUE;
+
+	if ((opcode >= BT_HCI_OP_LE_SET_ADV_PARAM) &&
+		(opcode <= BT_HCI_OP_LE_CREATE_CONN)) {
```

Comment:
Minor: the split line is incorrectly aligned (should be N tabs + 0-7 spaces to align with the appropriate opening parenthesis on the first line). That said, clang-format is actually suggesting that you could just merge these into a single line.

Reply:
Maybe using `IN_RANGE` from `zephyr/sys/util.h` would be more appropriate instead.